src/validate/mod.rs

```diff
@@ -15,7 +15,7 @@ lazy_static! {
         "^https?://[A-Za-z0-9-]+(?:\\.[A-Za-z0-9-]+)*(?::[0-9]+)?/(?:[A-Za-z0-9-]+/)*$"
     ).unwrap();
     static ref EMAIL_ADDRESS_FORMAT: Regex = Regex::new(
-        "^[a-z0-9\\-!#\\$%\\&`\\*\\+/=\\?\\^{\\|}~]+@[a-z0-9-]+(?:\\.[a-z0-9-]+)+$"
+        "^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,253}[a-zA-Z0-9])?)+$"
```

Comment:
If we're going to permit uppercase in the API, we need to make sure that we normalise the case before getting and setting bounce records in the db. Otherwise a bounce to `FOO@example.com` will not be matched by a query against `foo@example.com`.

Reply:
Alright, should I just fix it in this PR or should we open another issue about this? 

I feel this should be alright, Rust has a `to_lowercase` method (https://doc.rust-lang.org/std/string/struct.String.html#method.to_lowercase).

Reply:
If you're happy with it, I think we should just fix it in this PR.

Reply:
Also need ba test for it

Reply:
Hey guys, so I thought about just adding the `to_lowercase` function to to the bounces code, but it seemed cleaner to just make the email lowercase on deserialize (https://github.com/mozilla/fxa-email-service/pull/86/commits/37e8bbb8ae222930ba8c27bc976f2e4924107ac8#diff-170f5e597783d14eb65abe649bb9d4d2R45). 

Let me know if my thinking was right :)